runtime/minor_gc.c

```diff
@@ -622,6 +624,7 @@ void caml_empty_minor_heap_promote (caml_domain_state* domain, int participating
     if (Is_block(*v) && Is_young(*v)) {
       if (get_header_val(*v) == 0) { /* value copied to major heap */
```

Comment:
I actually think we need to do `caml_adjust_gc_speed(elt->mem, elt->max);` here after `if (Is_block(*v) && Is_young(*v)) {`.

In its current location, we will only apply the adjustment if the custom block has been promoted by another domain or root. We are promoting all the custom blocks so should do the adjustment for every element that is a block.

Reply:
Right, that does make sense, thanks for spotting this.